.circleci/config.yml

```diff
@@ -0,0 +1,137 @@
+
+version: 2
+jobs:
+  build:
+    docker:
+      # openjdk provides Scala dependencides. We may want to consider rolling
+      # our own image to also include JavaScript dependencies. Would speed up CI
+      # at the cost of maintaining a Docker image.
+      - image: openjdk:8
+        environment:
+          CI_TERRIBLENESS: 30.seconds
+        # Run two containers, $CIRCLE_NODE_TOTAL = 2, $CIRCLE_NODE_INDEX = [0|1]
+        parallelism: 2
+    working_directory: ~/linkerd
+    # All run steps execute in both containers unless otherwise specified.
+    # Deploy steps execute on container 0.
+    steps:
+      # Check out linkerd source code into ~/linkerd.
+      - checkout
+
+      # This allows us to run docker commands Since we're already in a docker
+      # container, this step sets up a remote docker environment for us.
+      - setup_remote_docker
+
+      # Restore build dependencies and downloaded files from the save_cache
+      # step. We use LinkerdBuild.scala as a proxy for Scala dependency changes,
+      # and package.json as a proxy for JavaScript dependency changes.
+      - restore_cache:
+          key: linkerd-{{ checksum "project/LinkerdBuild.scala" }}-{{ checksum "admin/src/main/resources/io/buoyant/admin/package.json" }}
```

Comment:
unfortunately, LinkerdBuild isn't really a good proxy for dependency changes. All of the files in files in projects basically need to be accounted for....